tests/README.md

```diff
@@ -1,25 +1,242 @@
+            - [Install the Python Dependencies from the Requirements File](#install-the-python-dependencies-from-the-requirements-file)
+        - [Running the Tests](#running-the-tests)
+    - [Modifying the Docker Images Employed](#modifying-the-docker-images-employed)
+        - [Example -- Running with a Custom Backend Service](#example----running-with-a-custom--backend-service)
+        - [Example -- Running with a Custom Client](#example----running-with-a-custom-client)
+    - [Running the Tests in a VM](#running-the-tests-in-a-vm)
+        - [Vagrant](#vagrant)
+    - [Known issues](#known-issues)
+        - [SSH](#ssh)
+        - [OS X](#os-x)
+    - [Tips and Tricks](#tips-and-tricks)
 
-Running tests is straight forward, just execute `bash run.sh`
+-------------------------------------------------------------------------------
 
-`run.sh` is simple wrapper that  grabs the latest test images, artifacts tools, and other test requirements. It also calls `py.test` with `--runfast` and `--runslow`, which runs all fast and slow tests.
 
-####OS X
+## Getting Started
 
-Currently, running integration tests on OS X is not straight forward due to: 
-https://github.com/docker/docker/issues/22753
+The dependencies for the integration tests are collected and organized in
+dependency files in the `./requirements` folder, and separated into
+
+
+> These requirement files are using the *hashicorp/bionic64* *Vagrant* image is
+> as the basis for the requirements files generated.
+
+| Debian                 | Alpine                 | Python                    |
+| :-------------:        | :-------------:        | :-----:                   |
+| *deb-requirements.txt* | *apt-requirements.txt* | *python-requirements.txt* |
+
+## Installing Dependencies
+
+### Debian
+
+```bash
+$ apt install -yyq $(cat requirements/deb-requirements.txt)
+```
+
+### Alpine Linux
+
+```bash
+$ apk --update add -yyq $(cat requirements/apk-requirements.txt)
+```
+
+### Python2 and Python3
+
+```bash
+$ pip  install -r requirements/python-requirements.txt
+$ pip3 install -r requirements/python-requirements.txt
+```
+
+
+-------------------------------------------------------------------------------
+
+
+## Running the Tests Locally
+
+> The tests can be run locally without any further involvement as long as all
+> the dependencies have been installed and are at the correct version. However,
+> managing dependencies, especially with Python can be a hassle. Therefore it is
+> recommended to add a virtual Python environment to isolate the dependencies
+> needed for running the integration tests.
+
+### Isolating Python Dependencies
+
+In order to avoid dependency mismanagement due to Python packages differing from
+one test environment to the other, it is recommended to use a python virtual
+environment. The "de-facto" standard is
+[virtualenv](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/).
```

Comment:
I think this should be before the `Installing Dependencies` section. If I am a new reader, I will most likely have installed already all pip stuff when I get to this point. Another reason is that below in this section we have duplicated the `pip install -r ...` command.

What do you think? We could have this first and still pointing out that it is optional and the reader might decide to go directly to dependencies installation to get python stuff natively installed.